### PR TITLE
Fix /improved tag write

### PR DESCRIPTION
**Pull request by wagnered**
1. Added check for existance of metaflac and vorbiscomments before writing flac or oga tags.
2. getID3 library doesn't remove pictures, only vorbiscomments.  Added work-around to replace existing file images in flac or oga files.
3.  Added missing constant `WRITE_ID3_ART` to `src/Config/ConfigurationKeyEnum.php`

This PR hasn't been tested on Windows: Specifically the Windows code in Vainfo.php.  **Hopefully someone can test PR in Windows environment before merging**.

### CODE REVIEW

**Comment by wagnered** [comment]
I broke it trying to please Code Climate.

File: src/Module/Util/VaInfo.php [hunk 1]
```diff
@@ -316,6 +316,36 @@ public function write_id3($tagData)
...
+                    return;
+                }
+            }
+
+            if (GETID3_OS_ISWINDOWS) {
+                $command = 'metaflac.exe --remove --block-type=PICTURE ' . escapeshellarg($this->filename);
+            } else {
+                $command = 'metaflac --remove --block-type=PICTURE ' . escapeshellarg($this->filename);
+            }
+            $commandError = `$command`;
```
Comment (lachlan-00) [inline src/Module/Util/VaInfo.php:347]: Happy with everything but is this variable used? it's coming up as dead code in scrutinizer

**Comment by lachlan-00** [comment]
> I broke it trying to please Code Climate.

i didn't read this before checking scrutinizer. was there something you needed to fix?

**Comment by lachlan-00** [comment]
i also only put the code checks on there as a warning and not a requirement. pleasing code climate is an almost impossible task

File: src/Module/Util/VaInfo.php (hunk 1 above)
Comment (wagnered) [reply src/Module/Util/VaInfo.php:347]: if you are referring to the variable `$command`, yes. 

See line 347: ``$commandError = `$command`;``    The backticks indicate that the metaflac command is being issued to the system to remove the pictures as a work-around.  It's not dead code.

**Comment by wagnered** [comment]
One of the lines in Art.php got split at the wrong place.  I'm surprised that php didn't give a run-time error. Have fixed it and tested it , and am in the process of cleaning up the code and pushing it to my fork.

**Comment by wagnered** [comment]
The main complaint of Code Climate now is "Define a constant instead of duplicating this literal "picturetypeid" 10 times." And a bunch of other literals.   To me it's "6 of one or a half-dozen of the other."

It would be nice to know if the changes work on Windows.  Other than that It should be ready, from my perspective.

File: src/Repository/Model/Catalog.php [hunk 2]
```diff
@@ -2933,10 +2933,10 @@ public static function process_action($action, $catalogs, $options = null)
                 $catalogs = self::get_catalogs();
                 // Intentional break fall-through
             case 'update_file_tags':
-                $write_id3 = AmpConfig::get('write_id3', false);
-                AmpConfig::set('write_id3', 'true', true);
+                $write_id3     = AmpConfig::get('write_id3', false);
                 $write_id3_art = AmpConfig::get('write_id3_art', false);
-                AmpConfig::set('write_id3_art', 'true', true);
+                AmpConfig::set_by_array(['write_id3' => 'true'], true);
+                AmpConfig::set_by_array(['write_id3_art' => 'true'], true);
```
Comment (usox) [inline src/Repository/Model/Catalog.php:2939]: After merging this, I'll check if it's possible to drop those temporary config settings and replace them by some parameters for the tagwriter

**Comment by wagnered** [comment]
From my perspective, I think it's ready to be merged.

**Pull request merged**